lua/intro.lua

```diff
@@ -0,0 +1,149 @@
+local function show_intro()
+    -- Don't show the intro if a buffer is loaded
+    if vim.api.nvim_buf_line_count(0) > 1 or vim.api.nvim_buf_get_name(0) ~= "" then
+        return
+    end
+
+    -- Create a new unlisted buffer
+    local buf = vim.api.nvim_create_buf(false, true)
+    local version_string = vim.api.nvim_exec2("version", { output = true }).output
+    -- we only need the first line of the version
+    version_string = version_string:sub(2, version_string:find("\n", 2) - 1)
+    local version = vim.version()
+
+    local lines = {
+        version_string,
+        "",
+        "Nvim is open source and freely distributable",
+        "https://neovim.io/#chat",
+        "",
+        "type  :help nvim<Enter>       if you are new! ",
+        "type  :checkhealth<Enter>     to optimize Nvim",
+        "type  :q<Enter>               to exit         ",
+        "type  :help<Enter>            for help        ",
+        "",
+        string.format("type  :help news<Enter> to see changes in v%d.%d", version.major, version.minor),
+        "",
+    }
+
+    local help_messages = {
+        {
+            "Sponsor Vim development!",
+            "type  :help sponsor<Enter>    for information ",
+        },
+        {
+            "Become a registered Vim user!",
+            "type  :help register<Enter>   for information ",
+        },
+        {
+            "Help poor children in Uganda!",
+            "type  :help iccf<Enter>       for information ",
+        },
+    }
+
+    -- Show the sponsor and register message one out of four times, the Uganda
+    -- message two out of four times.
+    local help_type = math.random(4)
```

Comment:
I've never spotted any other message in `help_messages` than the Uganda one. Possibly that stems from `math.random` in Neovim's lua inheriting the C semantics (`:help math.random`):

  ```help
  math.random([{m} [, {n}]])                                       *math.random()*
          This function is an interface to the simple pseudo-random generator
          function `rand` provided by ANSI C. (No guarantees can be given for
          its statistical properties.)
  
          When called without arguments, returns a pseudo-random real number in
          the range `[0,1)`. When called with a number {m}, `math.random`
          returns a pseudo-random integer in the range `[1, m]`. When called
          with two numbers {m} and {n}, `math.random` returns a pseudo-random
          integer in the range `[m, n]`.
  ```

  Which means that `math.random`, just like C's `rand`, returns the same sequence on startup unless seeded (see [cppreference on `rand`][cppreference]. And that happens to be `4` at least in my testings. [cppreference] also lists a usable workaround: seeding the RNG by time before sampling.

[cppreference]: https://en.cppreference.com/w/c/numeric/random/rand

```suggestion
    -- math.random returns the same on every startup if not seeded explicitly
    math.randomseed(os.time())
    local help_type = math.random(4)
```

Reply:
Yes, I was doing almost all the testing without closing Neovide, so I did not notice. I was also confused with the documentation of the built in rand which says an internal seed is used.

I will fix

Reply:
Ok fixed.

I would have hoped that Neovim set the random seed itself though, since now each plugin has to initialize it separately and possibly make the random numbers less random in the progress.

Reply:
I did a few test runs and this is what I got
```
uuuuuuuuuuuuuuuuuuuuuuuu
rrrrrrrrrrrrr
ssssssssssssssss
```
So it seems to be pretty decent.

Reply:
I'm not exactly sure how desirable non-deterministic random numbers inside an editor really are anyway. Thank you either way!